### PR TITLE
Fixing test suite

### PR DESCRIPTION
**Pull request by Nek-**
Hello there, trying to help to move forward.

This PR removes the test suite for PHP 8.0 because the dev dependencies are just not compatible with PHP 8.0.

It updates Sylius version to 1.10 and fixes behat tests.

It also allow to update to Symfony 5.3 thanks to the removal of the usage of `symfony/event-dispatcher-contracts` in favor of `symfony/contracts`.

Work is split in many commits to help you review with ease.

This PR targets 2.x but I plan to make it also on 3.x branch. Please send feedbacks!

Feel free to add `hacktoberfest-accepted` to help me complete my hacktoberfest challenge 😊 .

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -24,7 +24,7 @@
         "symfony/config": "^4.4 || ^5.0",
         "symfony/dependency-injection": "^4.4 || ^5.0",
         "symfony/event-dispatcher": "^4.4 || ^5.0",
-        "symfony/event-dispatcher-contracts": "^1.1",
+        "symfony/contracts": "^v1.1.10 || ^2.4.0",
```
Comment (loevgaard) [inline composer.json]: Why is this change needed?

**Comment by loevgaard** [comment]
Thank you for this, @Nek-. The v2 of this plugin will not be maintained a short period after v3 is released (at least only security fixes), so if you see any improvements on v3 we would appreciate this also :)

File: composer.json (hunk 1 above)
Comment (Nek-) [reply composer.json]: This is a dependency of the package [symfony/symfony](https://packagist.org/packages/symfony/symfony) and stands as replacement for `symfony/event-dispatcher-contracts`

image

Maybe I missread the situation and using `symfony/event-dispatcher-contracts` with constraint `^1.0 || ^2.0` is fine. Composer was suggesting using `symfony/contracts` instead so I did...

**Comment by Nek-** [comment]
@loevgaard the v3 has some issues in common (ie test suite not compatible with PHP 8 because of unused dev dependencies or too old dependencies (like sylius 1.7)) this is why I'm looking into updating both versions. But there's also the tag bundle to update (I'll send some work tomorrow about it).

I see the test suite is not passing yet, be sure I will fix it!

File: composer.json (hunk 1 above)
Comment (Nek-) [reply composer.json]: https://github.com/Setono/SyliusAnalyticsPlugin/pull/57 was working great! I removed my change for that.

**Comment by Nek-** [comment]
I am sorry I do not understand why psalm do not pass on the CI and I can't suppress issue with ECS... I really do not understand how this package work. It seems like it should work with the following configuration, but without having it as a dependency I can't understand how it could work.

```php
<?php

declare(strict_types=1);

use Symfony\Component\DependencyInjection\Loader\Configurator\ContainerConfigurator;

return static function (ContainerConfigurator $containerConfigurator): void {
    $containerConfigurator->import(__DIR__ . '/vendor/setono/coding-standard/easy-coding-standard.php');
};
```

☝️ missing on this package to allow ECS to work? But how can it work without setono/coding-standard as dependency ? But this package conflicts with sylius/coding-standard ?!

**Comment by Nek-** [comment]
My hotfix can be removed: https://github.com/symfony/symfony/pull/43801

**Comment by Nek-** [comment]
@loevgaard I SEE GREEN. 🙈  (finally)

_commits are all annotated with details_

File: src/DependencyInjection/Configuration.php [hunk 2]
```diff
@@ -20,16 +20,14 @@ public function getConfigTreeBuilder(): TreeBuilder
...
 
         /** @psalm-suppress PossiblyUndefinedMethod,PossiblyNullReference */
         $rootNode
             ->addDefaultsIfNotSet()
             ->children()
-                ->scalarNode('driver')->defaultValue(SyliusResourceBundle::DRIVER_DOCTRINE_ORM)->cannotBeEmpty()->end()
-            ->end()
-        ;
+                ->scalarNode('driver')->defaultValue(SyliusResourceBundle::DRIVER_DOCTRINE_ORM)->cannotBeEmpty()
+        ; // Note: `->end()->end()` are removed since they do nothing and are dead code detected by psalm
```
Comment (loevgaard) [inline src/DependencyInjection/Configuration.php]: Just remove these lines :)

File: src/DependencyInjection/Configuration.php [hunk 3]
```diff
@@ -55,14 +53,15 @@ private function addResourcesSection(ArrayNodeDefinition $node): void
...
-            ->end()
+                                        ->scalarNode('form')->defaultValue(PropertyType::class)->cannotBeEmpty() // ->end()
+//                                    ->end()
+//                                ->end()
+//                            ->end()
+//                        ->end()
+//                    ->end()
+//                ->end()
+//            ->end()
+        // statements `->end()` is dead code correctly detected by psalm
```
Comment (loevgaard) [inline src/DependencyInjection/Configuration.php]: Just remove these lines :)

File: src/EventListener/AddLibrarySubscriber.php [hunk 4]
```diff
@@ -23,7 +23,7 @@ public function add(RequestEvent $event): void
     {
         $request = $event->getRequest();
 
-        if (!$event->isMasterRequest() || !$this->isShopContext($request)) {
+        if (!$this->isMainRequest($event) || !$this->isShopContext($request)) {
```
Comment (loevgaard) [inline src/EventListener/AddLibrarySubscriber.php:26]: Doesn't work with SF4.4. You should revert these

File: src/EventListener/TagSubscriber.php [hunk 5]
```diff
@@ -73,4 +74,15 @@ protected function isShopContext(Request $request = null): bool
...
     }
+
+    protected function isMainRequest(RequestEvent $event): bool
+    {
+        if (method_exists($event, 'isMainRequest')) {
+            return $event->isMainRequest();
+        }
+
+        // BC Layer for Symfony < 5.3
+        /** @psalm-suppress DeprecatedMethod */
```
Comment (loevgaard) [inline src/EventListener/TagSubscriber.php:85]: Add it to Psalm config instead

File: tests/Application/config/packages/lexik_jwt_authentication.yaml [hunk 6]
```diff
@@ -0,0 +1,4 @@
+lexik_jwt_authentication:
+    secret_key: '%env(resolve:JWT_SECRET_KEY)%'
+    public_key: '%env(resolve:JWT_PUBLIC_KEY)%'
+    pass_phrase: '%env(JWT_PASSPHRASE)%'
```
Comment (loevgaard) [inline tests/Application/config/packages/lexik_jwt_authentication.yaml]: Not sure we need a password env var

**Comment by loevgaard** [comment]
Just a few comments, else it's a good job 🎉

File: src/EventListener/AddLibrarySubscriber.php (hunk 4 above)
Comment (Nek-) [reply src/EventListener/AddLibrarySubscriber.php:26]: It does! You probably misread it, it's a BC layer (by using a private method).

File: src/EventListener/TagSubscriber.php (hunk 5 above)
Comment (Nek-) [reply src/EventListener/TagSubscriber.php:85]: How? (+ I actually think it's nice to have this comment to make it easy to drop the bc layer later)

File: src/DependencyInjection/Configuration.php (hunk 2 above)
Comment (Nek-) [reply src/DependencyInjection/Configuration.php]: Removed!

File: src/DependencyInjection/Configuration.php (hunk 3 above)
Comment (Nek-) [reply src/DependencyInjection/Configuration.php]: Removed!

File: tests/Application/config/packages/lexik_jwt_authentication.yaml (hunk 6 above)
Comment (Nek-) [reply tests/Application/config/packages/lexik_jwt_authentication.yaml]: Removed :) . (cute catch ^^)

**Comment by Nek-** [comment]
@loevgaard updated! :)

File: src/EventListener/TagSubscriber.php (hunk 5 above)
Comment (loevgaard) [reply src/EventListener/TagSubscriber.php:85]: Yeah, I think you will gain the same benefits of just removing it from Psalm config, but let's just keep it :)

**Comment by loevgaard** [comment]
Thank you, @Nek-. Great job 🎉

**Comment by Nek-** [comment]
@loevgaard do you want me to forward commits to 3.x branch?

**Comment by loevgaard** [comment]
@Nek-: `3.x` is basically a whole other plugin because we switched from client side to server side in that branch, but anything is welcome as always :)
